diff --git a/node_modules/react-native/React/Base/RCTBridgeDelegate.h b/node_modules/react-native/React/Base/RCTBridgeDelegate.h
index 2ae1b8b..483ef48 100644
--- a/node_modules/react-native/React/Base/RCTBridgeDelegate.h
+++ b/node_modules/react-native/React/Base/RCTBridgeDelegate.h
@@ -77,4 +77,6 @@
  */
 - (NSDictionary<NSString *, Class> *)extraLazyModuleClassesForBridge:(RCTBridge *)bridge;

+- (void)bridgeDidInitializeJSGlobalContext:(void *)contextRef;
+
 @end
diff --git a/node_modules/react-native/React/CxxBridge/RCTCxxBridge.mm b/node_modules/react-native/React/CxxBridge/RCTCxxBridge.mm
index 18b0e09..f83c283 100644
--- a/node_modules/react-native/React/CxxBridge/RCTCxxBridge.mm
+++ b/node_modules/react-native/React/CxxBridge/RCTCxxBridge.mm
@@ -362,6 +362,10 @@ - (void)start
   dispatch_group_enter(prepareBridge);
   [self ensureOnJavaScriptThread:^{
     [weakSelf _initializeBridge:executorFactory];
+
+    facebook::jsi::Runtime &runtime = *(facebook::jsi::Runtime *)weakSelf.runtime;
+    [weakSelf.delegate bridgeDidInitializeJSGlobalContext:runtime.context()];
+
     dispatch_group_leave(prepareBridge);
   }];

diff --git a/node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp b/node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp
index a6e59fb..cac57f6 100644
--- a/node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp
+++ b/node_modules/react-native/ReactCommon/jsi/JSCRuntime.cpp
@@ -51,6 +51,8 @@ class JSCRuntime : public jsi::Runtime {
       const std::string& sourceURL) override;
   jsi::Object global() override;

+  void * context() override;
+
   std::string description() override;

   bool isInspectable() override;
@@ -403,6 +405,10 @@ jsi::Object JSCRuntime::global() {
   return createObject(JSContextGetGlobalObject(ctx_));
 }

+void * JSCRuntime::context() {
+  return ctx_;
+}
+
 std::string JSCRuntime::description() {
   if (desc_.empty()) {
     desc_ = std::string("<JSCRuntime@") + to_string(this) + ">";
diff --git a/node_modules/react-native/ReactCommon/jsi/jsi/jsi.h b/node_modules/react-native/ReactCommon/jsi/jsi/jsi.h
index 33e3008..835f517 100644
--- a/node_modules/react-native/ReactCommon/jsi/jsi/jsi.h
+++ b/node_modules/react-native/ReactCommon/jsi/jsi/jsi.h
@@ -188,6 +188,8 @@ class Runtime {
   /// \return the global object
   virtual Object global() = 0;

+  virtual void *context() = 0;
+
   /// \return a short printable description of the instance.  This
   /// should only be used by logging, debugging, and other
   /// developer-facing callers.
